lib/services.js

```diff
@@ -6,27 +6,30 @@ import { InstallationProxyService, INSTALLATION_PROXY_SERVICE_NAME } from './ins
 import { AfcService, AFC_SERVICE_NAME } from './afc';
 import { NotificationProxyService, NOTIFICATION_PROXY_SERVICE_NAME } from './notification-proxy';
 import { HouseArrestService, HOUSE_ARREST_SERVICE_NAME } from './house-arrest';
-import { InstrumentService, INSTRUMENT_SERVICE_NAME_VERSION_14, INSTRUMENT_SERVICE_NAME} from './instrument';
+import { InstrumentService, INSTRUMENT_SERVICE_NAME_VERSION_14, INSTRUMENT_SERVICE_NAME } from './instrument';
 import { TestmanagerdService, TESTMANAGERD_SERVICE_NAME_VERSION_14, TESTMANAGERD_SERVICE_NAME } from './testmanagerd';
-import { MCInstallProxyService, MC_INSTALL_SERVICE_NAME} from './mcinstall';
+import { MCInstallProxyService, MC_INSTALL_SERVICE_NAME } from './mcinstall';
+import { ImageMounter, MOBILE_IMAGE_MOUNTER_SERVICE_NAME } from './imagemounter';
 import PlistService from './plist-service';
 import semver from 'semver';
+import { findDeveloperImage } from './imagemounter/utils/list_developer_image';
+import log from './logger';
 
 const CRASH_LOG_SERVICE_NAME = 'com.apple.crashreportcopymobile';
 const INSTRUMENT_HANDSHAKE_VERSION = 14;
 const TESTMANAGERD_HANDSHAKE_VERSION = 14;
 
-async function startSyslogService (udid, opts = {}) {
+async function startSyslogService(udid, opts = {}) {
   const socket = await startService(udid, SYSLOG_SERVICE_NAME, opts.socket);
```

Comment:
That inspires me. We can make it as a property as opts, as well as the repo/url/localPath of developer image & signature?
Also, what if we check mount status before start service rather than mount it after failure?

Reply:
yea, could be.

Hm... good point. Personally, I'd like to do it in case of failure as the current way. If the environment already has expected images, searching the dmg etc may be just an extra delay. (the delay should be very small tho, I know of.). Or we can define a device image check separately instead of inside `startService`. So,this appium-ios-device expects the caller to call `autoMountDeveloperImage` as their responsibility.

So, appium xcuitest driver will call `autoMountDeveloperImage` first explicitly in a new session request. After the check, the session won't do the image check as same as current appium-ios-device behavior.

Not correct implementation, but:
```
# in xcuitest

function startSession() {
   if (this.isRealDevice() && checkDeviceImageCapability) {
     // do only once in early new session request
      autoMountDeveloperImage()
   }
   startSyslogService(); # etc won't call `autoMountDeveloperImage` inside startService.
}

```

Reply:
Hm, it seems like I like the explicit way. (after writing the above comment).

So, the `startService` won't call `autoMountDeveloperImage` inside the method. Instead, the caller (appium-xcuitest-driver in Appium case) should call `autoMountDeveloperImage` explicitly to check the local device images.

Reply:
I've changed this method into `utilities` and make user decides which github repo or local path to use. But this makes `startImageMounterService` and `moutDeveloperImage` kind of redundant. 
Maybe it should just look for github image and download&unzip, return the path, and let user use service to mount images. But this is unlikely to be a device related procedure which feels weird to be in this library in this way. I'm confused now🤔

Reply:
> Maybe it should just look for github image and download&unzip, return the path, and let user use service to mount images. But this is unlikely to be a device related procedure which feels weird to be in this library in this way. I'm confused now🤔

This makes sense and the logic seems more flexible to me. Just add a proper steps and maybe a code example into the module's README file

Reply:
> 

Actually I think we are truely lack of code example for many parts😁.